source/store/AppContext.js

```diff
@@ -0,0 +1,37 @@
+import PropTypes from 'prop-types';
+import React, { useState } from 'react';
+import env from 'react-native-config';
+import StorageService, { APP_ENV_KEY } from '../services/StorageService';
+
+const AppContext = React.createContext();
+
+function AppProvider({ children }) {
+  const acceptedModes = ['development', 'production'];
+  const defaultMode = env.APP_ENV || acceptedModes[0];
+  const [mode, setMode] = useState(defaultMode);
+
+  const handleSetMode = (newMode) => {
+    if (acceptedModes.includes(newMode)) {
+      setMode(newMode);
+    }
+  };
+
+  const provider = {
+    mode,
+    handleSetMode: (newMode) => handleSetMode(newMode),
+    isDevMode: mode === 'development',
+  };
+
+  React.useEffect(() => {
```

Comment:
Very minor, but we usually import useEffect, and you already import useState, so for style it's nicer to import useEffect and not write React. here.

Reply:
Fixed!